src/code/z_message.c

```diff
@@ -1,24 +1,167 @@
+
+    msgCtx->unk1204A[0] = 0xBD;
+    msgCtx->unk1204A[1] = 0xB8;
+    msgCtx->unk1204A[2] = 0xB3;
+    msgCtx->unk1204A[3] = 0xAE;
+    msgCtx->unk1204A[4] = 0xA9;
+    func_80147520();
+    D_801F6B0C = 0x50;
+    D_801F6B10 = 0x96;
+    D_801F6B0E = 0xFF;
+    D_801F6B12 = 0xA;
+    D_801F6B16 = 0xA;
+    D_801F6B14 = 0xA;
+    D_801F6B18 = 0xFF;
+    D_801F6B1C = 0xFF;
+    D_801F6B1A = 0x32;
+    D_801F6B1E = 0xA;
+    D_801F6B22 = 0xA;
+    D_801F6B20 = 0xA;
+}
+
+s32 Message_ShouldAdvance(GlobalContext* globalCtx) {
+    MessageContext* msgCtx = &globalCtx->msgCtx;
+    Input* controller = CONTROLLER1(globalCtx);
+
+    if ((msgCtx->unk12020 == 0x10) || (msgCtx->unk12020 == 0x11)) {
+        if (CHECK_BTN_ALL(controller->press.button, BTN_A)) {
+            play_sound(NA_SE_SY_MESSAGE_PASS);
+        }
+        return CHECK_BTN_ALL(controller->press.button, BTN_A);
+    } else {
+        if (CHECK_BTN_ALL(controller->press.button, BTN_A) || CHECK_BTN_ALL(controller->press.button, BTN_B) ||
+            CHECK_BTN_ALL(controller->press.button, BTN_CUP)) {
+            play_sound(NA_SE_SY_MESSAGE_PASS);
+        }
+        return CHECK_BTN_ALL(controller->press.button, BTN_A) || CHECK_BTN_ALL(controller->press.button, BTN_B) ||
+               CHECK_BTN_ALL(controller->press.button, BTN_CUP);
+    }
+}
+
+s32 Message_ShouldAdvanceSilent(GlobalContext* globalCtx) {
+    MessageContext* msgCtx = &globalCtx->msgCtx;
+    Input* controller = CONTROLLER1(globalCtx);
+
+    if (msgCtx->unk12020 == 0x10 || msgCtx->unk12020 == 0x11) {
+        return CHECK_BTN_ALL(controller->press.button, BTN_A);
+    } else {
+        return CHECK_BTN_ALL(controller->press.button, BTN_A) || CHECK_BTN_ALL(controller->press.button, BTN_B) ||
+               CHECK_BTN_ALL(controller->press.button, BTN_CUP);
+    }
+}
+
+void func_801477B4(GlobalContext* globalCtx) {
+    MessageContext* msgCtx;
+
+    msgCtx = &globalCtx->msgCtx;
+    if (globalCtx->msgCtx.unk11F10 != 0) {
+        msgCtx->unk12023 = 2;
+        msgCtx->msgMode = 0x43;
+        msgCtx->unk12020 = 0;
+        play_sound(NA_SE_PL_WALK_GROUND - SFX_FLAG);
+    }
+}
 
 #pragma GLOBAL_ASM("asm/non_matchings/code/z_message/func_80147818.s")
 
 #pragma GLOBAL_ASM("asm/non_matchings/code/z_message/func_80147F18.s")
 
 #pragma GLOBAL_ASM("asm/non_matchings/code/z_message/func_80148558.s")
 
+#ifdef NON_MATCHING
+void func_80148B98(GlobalContext* globalCtx, u8 arg1) {
+    static s16 held = 0;
+    MessageContext* msgCtx = &globalCtx->msgCtx;
+    Input* curInput = CONTROLLER1(globalCtx);
+
+    if ((curInput->rel.stick_y > 29) && held == 0) {
+        held = 1;
+        msgCtx->choiceIndex--;
+        if (msgCtx->choiceIndex > 128) {
+            msgCtx->choiceIndex = 0;
+        } else {
+            play_sound(NA_SE_SY_CURSOR);
+        }
+        return;
+    }
+
+    else if ((curInput->rel.stick_y < -29) && held == 0) {
+        held = 1;
+        msgCtx->choiceIndex++;
```

Comment:
```suggestion
        return;
    } else if ((curInput->rel.stick_y < -29) && held == 0) {
        held = 1;
        msgCtx->choiceIndex++;
```

Reply:
done in https://github.com/zeldaret/mm/commit/eaa4f44462da18c9c81ba828fca65470abf3b526